models/user_mail_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetEmailAddresses(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	emails, _ := GetEmailAddresses(int64(1))
+	assert.Len(t, emails, 3)
+	assert.False(t, emails[0].IsPrimary)
+	assert.True(t, emails[2].IsActivated)
+	assert.True(t, emails[2].IsPrimary)
+
+	emails, _ = GetEmailAddresses(int64(2))
+	assert.Len(t, emails, 2)
+	assert.True(t, emails[0].IsPrimary)
+	assert.True(t, emails[0].IsActivated)
+}
+
+func TestIsEmailUsed(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	isExist, _ := IsEmailUsed("")
+	assert.True(t, isExist)
+	isExist, _ = IsEmailUsed("user11@example.com")
+	assert.True(t, isExist)
+	isExist, _ = IsEmailUsed("user1234567890@example.com")
+	assert.False(t, isExist)
+}
+
+func TestAddEmailAddress(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	assert.NoError(t, AddEmailAddress(&EmailAddress{
+		Email:       "user1234567890@example.com",
+		IsPrimary:   true,
+		IsActivated: true,
+	}))
+
+	// ErrEmailAlreadyUsed
+	err := AddEmailAddress(&EmailAddress{
+		Email: "user1234567890@example.com",
+	})
+	assert.Error(t, err)
+	assert.True(t, IsErrEmailAlreadyUsed(err))
+
+	// insert multiple email address
+	emails := make([]*EmailAddress, 2)
+	emails[0] = &EmailAddress{
+		Email:       "user1234@example.com",
+		IsActivated: true,
+	}
+	emails[1] = &EmailAddress{
+		Email:       "user5678@example.com",
+		IsActivated: true,
+	}
+	assert.NoError(t, AddEmailAddresses(emails))
```

Comment:
Not to be nitpicky, but this function tests both `AddEmailAddress(..)` and `AddEmailAddresses(..)`. Could we split it into two tests: `TestAddEmailAddress` and `TestAddEmailAddresses`?